libraries/AP_OpticalFlow/AP_OpticalFlow_UPFLOW.cpp

```diff
@@ -87,6 +259,91 @@ void AP_OpticalFlow_UPFLOW::init()
     }
     // open serial port with baud rate of 19200
     uart->begin(19200);
+
+    /*
+        LC306 or LC302-8B boards require initialization before use, but LC302-3C does not.
+        Initialization can divided into four parts
+
+        (0xAA)start --> (0xAB)internal parameter configuration
+        --> (0xBB)sensor parameter configuration (Loop) --> (0xDD)End
+            ^--------------------------------------/
+
+        | command sent | returned status |
+        | ------------ | --------------- |
+        |     0xAA     |      none       | 1. Start flag
+        | ------------ | --------------- |
+        |     0xAB     |      0xAB       | 2. internal parameter configuration flag
+        |     0x96     |                 | status code:  0x00 == success
+        |     0x26     |   status code   |               0x01 == IIC no response
+        |     0xbc     |                 |               0x02 == XOR error
+        |     0x50     |                 | send: 0xAB, 0x96, 0x26, 0xbc, 0x50, 0x5c(BCC without 0xAB)
+        |   BCC(0x5c)  | 0xAB xor status | success response: 0xAB, 0x00, 0xAB
+        | ------------ | --------------- |
+        |     0xBB     |      0xBB       | 3. sensor parameter configuration
+        |     0xdc     |                 |    sending data in upflow_sensor_cfg
+        |    address   |   status code   | same status code above
+        |     data     |                 | send: 0xBB, 0xdc, 0xXX, 0xXX, 0xXX(BCC without 0xBB)
+        |     BCC      | 0xBB xor status | success response: 0xBB, 0x00, 0xBB
+        | ------------ | --------------- |
+        |     0xDD     |      none       | 4. Stop flag
+
+        !:  Step 3 will run repeatedly until the sending is complete.
+            Each time the module receives an 0xBB instruction, it will return a corresponding status.
+    */
+
+    // uint32_t nbytes = 0;
+    uint8_t bcc_sum = 0;
+    uint8_t recv_buf[3];
+
+    //delay 100ms before initialization
+    hal.scheduler->delay(100);
```

Comment:
Not really a fan of delays in the init, this is called from the main thread. However, there are a few existing examples in rangefinder...

Reply:
Thank you for your feedback. I understand your concern about latency in the main thread, but this is to ensure the sensor is initialized correctly. It is necessary to wait 100 ms for the sensor to initialize after powering up. However, do you have any suggestions or a more elegant approach?